### DISCUSSION TITLE
CLI - Use the right data type when saving global configuration values

### DISCUSSION THREAD

**Discussion by richard67** [joomla/joomla-cms#47055] [2. Cold Features]
### Is your feature request related to a problem? Please describe.

Currently the CLI command `config:set` changes all integer values to string values when saving changes, regardless whether these values were changed or not.

Values which seem to be boolean because they have content "true" or "false" are saved as booleans, but that is not really safe.

See the comment here: https://github.com/joomla/joomla-cms/pull/42618#issuecomment-1879638546 .

### Describe the solution you'd like

The CLI should save configuration in the same way as the administrator does, so the data types of the particular values are determined by the `administrator/components/com_config/forms/application.xml` file or by the existing `configuration.php` file.

The should **_not_** be a data type conversion based on the content of a value like it is currently done for boolean here https://github.com/joomla/joomla-cms/blob/5.0-dev/libraries/src/Console/SetConfigurationCommand.php#L445-L450 .

### Additional context

See also issue #42612 and PR #42618 .

**Comment by brianteeman** [comment]
Maybe I am testing this wrong but when I tried to use config:set to change the smtpport ONLY that field was changed from an integer to string. No other values were changed. Changing massmailoff from true to false didnt change to a boolean

**Comment by richard67** [comment]
Hmm, but the code which does that is still here: https://github.com/joomla/joomla-cms/blob/5.3-dev/libraries/src/Console/SetConfigurationCommand.php#L445-L450

**Comment by brianteeman** [comment]
As I said maybe my test is wrong 

<img width="845" height="194" alt="Image" src="https://github.com/user-attachments/assets/0899baa6-c327-4546-a955-7e38fe8f0e97" />

<img width="1053" height="765" alt="Image" src="https://github.com/user-attachments/assets/8eb19a3e-0d4f-459f-83bb-690c2189a0d9" />

**Comment by richard67** [comment]
@brianteeman Yes, the issue is vice versa. If you save a string value which has value `'true'` it might be saved as boolean, and that would be wrong. See my comment here: https://github.com/joomla/joomla-cms/pull/42618#issuecomment-1879638546

That means: Test if you can save e.g. `$smtphost` with a string value `'true'`. IF the result is saved as string: No issue. If it is saved as boolean: Issue.

**Comment by brianteeman** [comment]
now i am even more confused. 

>  changes all integer values to string values when saving changes, regardless whether these values were changed or not.

The only values that were changed were the values I instructed to be changed and yes the integers were changed to strings which i agree is wrong but the true/false did not change data type
